components/director/internal/authenticator/claims/validator.go

```diff
@@ -153,6 +177,77 @@ func (v *validator) validateRuntimeConsumer(ctx context.Context, claims Claims)
 	return tx.Commit()
 }
 
+func (v *validator) validateApplicationProvider(ctx context.Context, claims Claims) error {
+	tx, err := v.transact.Begin()
+	if err != nil {
+		log.C(ctx).Errorf("An error has occurred while opening transaction: %v", err)
+		return errors.Wrapf(err, "An error has occurred while opening transaction")
+	}
+	defer v.transact.RollbackUnlessCommitted(ctx, tx)
+
+	ctx = persistence.SaveToContext(ctx, tx)
+
+	if len(claims.TokenClientID) == 0 {
+		log.C(ctx).Errorf("Could not find consumer token client ID")
+		return apperrors.NewUnauthorizedError("could not find consumer token client ID")
+	}
+	if len(claims.Region) == 0 {
+		log.C(ctx).Errorf("Could not determine consumer token's region")
+		return apperrors.NewUnauthorizedError("could not determine token's region")
+	}
+
+	tokenClientID := strings.TrimPrefix(claims.TokenClientID, v.tokenPrefix)
+	filters := []*labelfilter.LabelFilter{
+		labelfilter.NewForKeyWithQuery(v.subscriptionProviderLabelKey, fmt.Sprintf("\"%s\"", tokenClientID)),
+		labelfilter.NewForKeyWithQuery(tenant.RegionLabelKey, fmt.Sprintf("\"%s\"", claims.Region)),
+	}
+
+	providerInternalTenantID := claims.Tenant[tenantmapping.ProviderTenantKey]
+	providerExternalTenantID := claims.Tenant[tenantmapping.ProviderExternalTenantKey]
+	ctxWithProviderTenant := tenant.SaveToContext(ctx, providerInternalTenantID, providerExternalTenantID)
+
+	log.C(ctx).Infof("Get application template in provider tenant %s for labels %s: %s and %s: %s", providerInternalTenantID, tenant.RegionLabelKey, claims.Region, v.subscriptionProviderLabelKey, tokenClientID)
+	applicationTemplate, err := v.appTemplateSvc.GetByFilters(ctxWithProviderTenant, filters)
+	if err != nil {
+		log.C(ctx).WithError(err).Errorf("Error while getting application template in provider tenant %s for labels %s: %s and %s: %s: %v", providerInternalTenantID, tenant.RegionLabelKey, claims.Region, v.subscriptionProviderLabelKey, tokenClientID, err)
+		return errors.Wrapf(err, "failed to get application template in tenant %s for labels %s: %s and %s: %s", providerInternalTenantID, tenant.RegionLabelKey, claims.Region, v.subscriptionProviderLabelKey, tokenClientID)
+	}
+	log.C(ctx).Infof("Found application template with ID %q in provider tenant %s for labels %s: %s and %s: %s", applicationTemplate.ID, providerInternalTenantID, tenant.RegionLabelKey, claims.Region, v.subscriptionProviderLabelKey, tokenClientID)
+
+	consumerInternalTenantID := claims.Tenant[tenantmapping.ConsumerTenantKey]
+	consumerExternalTenantID := claims.Tenant[tenantmapping.ExternalTenantKey]
+	ctxWithConsumerTenant := tenant.SaveToContext(ctx, consumerInternalTenantID, consumerExternalTenantID)
+
+	applicatinsFilter := []*labelfilter.LabelFilter{
```

Comment:
I think this filter is unnecessary

Reply:
Without it will look for apps in all subaccounts instead only in subscriber subaccount only. 
The idea is to get all apps in this subaccount and go over them to look for app that is based on the template from the provider.